Android build fails on RN 0.65 after updating toolbar-android to 0.2.0

Issue

> Task :react-native-community_toolbar-android:compileDebugJavaWithJavac FAILED

After updating React Native from 64.1 to 65.1, android build fails with the error below:

```
node_modules/@react-native-community/toolbar-android/android/src/main/java/com/reactnativecommunity/toolbarandroid/ReactToolbar.java:133: error: method does not override or implement a method from a supertype
    @Override
    ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-community_toolbar-android:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```

I updated toolbar-android to 0.2.0 since the previous version was causing issues as well.

As a current workaround, I removed the @ Override above the getExtras() method (that's where the issue is pointing at). Doing this fixes my issue and my app works without a problem. Not sure if it an issue with my environment or if anyone has had this issue before.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -10,7 +10,7 @@ defaults: &defaults
 linux_defaults: &linux_defaults
   <<: *defaults
   docker:
-  - image: circleci/node:8
+  - image: circleci/node:12
   environment:
   - PATH: "/opt/yarn/yarn-v1.5.1/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 
@@ -18,7 +18,7 @@ linux_defaults: &linux_defaults
 android_defaults: &android_defaults
   <<: *defaults
   docker:
-  - image: circleci/android:api-27-node8-alpha
+  - image: circleci/android:api-29
   resource_class: "medium"
   environment:
   - TERM: "dumb"
@@ -161,10 +161,6 @@ jobs:
         name: Build Android Example App (RN65)
         command: |-
           cd example/ToolbarAndroidExample-065/android
-          pwd
-          ls -l ../node_modules/
-          ls -l ../node_modules/@react-native-community/cli-platform-android
-          ls -l ../node_modules/@react-native-community/cli-platform-android/native_modules.gradle
           ./gradlew clean assembleDebug
     - save-cache: *save-gradle-cache
 

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -162,6 +162,8 @@ jobs:
         command: |-
           cd example/ToolbarAndroidExample-065/android
           pwd
+          ls -l ../node_modules/
+          ls -l ../node_modules/@react-native-community/cli-platform-android
           ls -l ../node_modules/@react-native-community/cli-platform-android/native_modules.gradle
           ./gradlew clean assembleDebug
     - save-cache: *save-gradle-cache

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -35,8 +35,8 @@ macos_defaults: &macos_defaults
     xcode: "10.1.0"
 
 # Example
-at-example-61: &at-example-61 
-  working_directory: ~/toolbar-android/example/ToolbarAndroidExample-061
+at-example-65: &at-example-65 
+  working_directory: ~/toolbar-android/example/ToolbarAndroidExample-065
 
 # -------------------------
 #         ALIASES
@@ -56,11 +56,11 @@ aliases:
 
 - &restore-gradle-cache
   keys:
-  - gradle-cache-{{ checksum "android/build.gradle" }}-{{ checksum "example/ToolbarAndroidExample-061/android/build.gradle" }}-{{ checksum "example/ToolbarAndroidExample-061/android/app/build.gradle" }}
+  - gradle-cache-{{ checksum "android/build.gradle" }}-{{ checksum "example/ToolbarAndroidExample-065/android/build.gradle" }}-{{ checksum "example/ToolbarAndroidExample-065/android/app/build.gradle" }}
 - &save-gradle-cache
   paths:
   - ~/.gradle
-  key: gradle-cache-{{ checksum "android/build.gradle" }}-{{ checksum "example/ToolbarAndroidExample-061/android/build.gradle" }}-{{ checksum "example/ToolbarAndroidExample-061/android/app/build.gradle" }}
+  key: gradle-cache-{{ checksum "android/build.gradle" }}-{{ checksum "example/ToolbarAndroidExample-065/android/build.gradle" }}-{{ checksum "example/ToolbarAndroidExample-065/android/app/build.gradle" }}
 
 # INSTALLATION
 - &yarn
@@ -105,9 +105,9 @@ jobs:
     - run: *yarn
     - save-cache: *save-yarn-cache
     - run:
-        name: Checkout Example (RN61) 
+        name: Checkout Example (RN65) 
         command: |-
-          cd example/ToolbarAndroidExample-061
+          cd example/ToolbarAndroidExample-065
           rm -rf node_modules
           yarn cache clean
 
@@ -146,7 +146,7 @@ jobs:
     - run: *tsc
 
 
-  example-61-android-compile:
+  example-65-android-compile:
     <<: *android_defaults
     steps:
     - attach_workspace:
@@ -158,26 +158,26 @@ jobs:
           yes | sdkmanager --licenses || exit 0
           yes | sdkmanager --update || exit 0
     - run:
-        name: Build Android Example App (RN61)
+        name: Build Android Example App (RN65)
         command: |-
-          cd example/ToolbarAndroidExample-061/android
+          cd example/ToolbarAndroidExample-065/android
           ./gradlew clean assembleDebug
     - save-cache: *save-gradle-cache
 
-  example-61-ios:
+  example-65-ios:
     <<: *macos_defaults
     steps:
     - attach_workspace:
         at: ~/toolbar-android
     - run:
-        name: Build iOS example app (RN61)
+        name: Build iOS example app (RN65)
         command: |-
-          cd example/ToolbarAndroidExample-061/ios 
+          cd example/ToolbarAndroidExample-065/ios 
           rm -rf Prods
           pod install
           cd ..
           yarn ios
-          # react-native run-ios --project-path example/ToolbarAndroidExample-061/ios
+          # react-native run-ios --project-path example/ToolbarAndroidExample-065/ios
 
 # -------------------------
 #        WORKFLOWS
@@ -199,7 +199,7 @@ workflows:
     - tsc:
         requires:
         - linux-checkout
-    - example-61-android-compile:
+    - example-65-android-compile:
         requires:
         - linux-checkout
 # Disabled until we have macOS containers enabled        

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -18,7 +18,7 @@ linux_defaults: &linux_defaults
 android_defaults: &android_defaults
   <<: *defaults
   docker:
-  - image: circleci/android:api-29
+  - image: circleci/android:api-29-node
   resource_class: "medium"
   environment:
   - TERM: "dumb"

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -23,7 +23,7 @@ android_defaults: &android_defaults
   environment:
   - TERM: "dumb"
   - ADB_INSTALL_TIMEOUT: 10
-  - _JAVA_OPTIONS: "-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap"
+  - _JAVA_OPTIONS: "-XX:+UnlockExperimentalVMOptions -XX:+UseContainerSupport"
   - GRADLE_OPTS: '-Dorg.gradle.daemon=false -Dorg.gradle.jvmargs="-XX:+HeapDumpOnOutOfMemoryError"'
   - BUILD_THREADS: 2
 

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -161,6 +161,8 @@ jobs:
         name: Build Android Example App (RN65)
         command: |-
           cd example/ToolbarAndroidExample-065/android
+          pwd
+          ls -l ../node_modules/@react-native-community/cli-platform-android/native_modules.gradle
           ./gradlew clean assembleDebug
     - save-cache: *save-gradle-cache
 

--- a/README.md
+++ b/README.md
@@ -27,6 +27,8 @@ npx pod-install
 react-native link @react-native-community/toolbar-android
 ```
 
+ Note: `@react-native-community/toolbar-android@0.2.0` supports `react-native@0.65` and up. For earlier version of react-native, please use `@react-native-community/toolbar-android@0.1.0-rc2` 
+ 
 ### Manual installation
 
 <details>

--- a/README.md
+++ b/README.md
@@ -27,8 +27,6 @@ npx pod-install
 react-native link @react-native-community/toolbar-android
 ```
 
- Note: `@react-native-community/toolbar-android@0.2.0` supports `react-native@0.65` and up. For earlier version of react-native, please use `@react-native-community/toolbar-android@0.1.0-rc2` 
- 
 ### Manual installation
 
 <details>

--- a/android/src/main/java/com/reactnativecommunity/toolbarandroid/ReactToolbar.java
+++ b/android/src/main/java/com/reactnativecommunity/toolbarandroid/ReactToolbar.java
@@ -130,7 +130,7 @@ public class ReactToolbar extends Toolbar {
       return null;
     }
 
-    @Override
+    // to be compatible with RN-0.65/fresco-2.5
     public Map<String, Object> getExtras() {
       return null;
     }

--- a/example/ToolbarAndroidExample-065/ToolbarAndroidExample/ToolbarAndroidExample.js
+++ b/example/ToolbarAndroidExample-065/ToolbarAndroidExample/ToolbarAndroidExample.js
@@ -39,7 +39,7 @@ export default class ToolbarAndroidExample extends Component<Props, State> {
       <View style={styles.container}>
         <ToolbarAndroid
           navIcon={require('./ic_menu_black_24dp.png')}
-          title="ToolbarAndroid Example"
+          title="ToolbarAndroid Example (RN-0.65)"
           style={styles.toolbar}
           actions={[
             {title: 'Button', icon: require('./relay.png'), show: 'always'},

--- a/example/ToolbarAndroidExample-065/__tests__/App-test.js
+++ b/example/ToolbarAndroidExample-065/__tests__/App-test.js
@@ -4,7 +4,7 @@
 
 import 'react-native';
 import React from 'react';
-import App from '../App';
+import App from '../ToolbarAndroidExample';
 
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';

--- a/example/ToolbarAndroidExample-065/android/app/.classpath
+++ b/example/ToolbarAndroidExample-065/android/app/.classpath
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11/"/>
+	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
+	<classpathentry kind="output" path="bin/default"/>
+</classpath>

--- a/example/ToolbarAndroidExample-065/android/app/.project
+++ b/example/ToolbarAndroidExample-065/android/app/.project
@@ -1,10 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>ToolbarAndroidExample</name>
-	<comment>Project android__ created by Buildship.</comment>
+	<name>ToolbarAndroidExample-app</name>
+	<comment>Project ToolbarAndroidExample-app created by Buildship.</comment>
 	<projects>
 	</projects>
 	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
 		<buildCommand>
 			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
 			<arguments>
@@ -12,11 +17,12 @@
 		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
 		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
 	</natures>
 	<filteredResources>
 		<filter>
-			<id>0</id>
+			<id>1630694511090</id>
 			<name></name>
 			<type>30</type>
 			<matcher>

--- a/example/ToolbarAndroidExample-065/android/app/.settings/org.eclipse.buildship.core.prefs
+++ b/example/ToolbarAndroidExample-065/android/app/.settings/org.eclipse.buildship.core.prefs
@@ -1,0 +1,2 @@
+connection.project.dir=..
+eclipse.preferences.version=1

--- a/example/ToolbarAndroidExample-expo-063/android/.project
+++ b/example/ToolbarAndroidExample-expo-063/android/.project
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>android_</name>
+	<name>ToolbarAndroidExampleexpo063</name>
 	<comment>Project android_ created by Buildship.</comment>
 	<projects>
 	</projects>
@@ -16,7 +16,7 @@
 	</natures>
 	<filteredResources>
 		<filter>
-			<id>1630614539693</id>
+			<id>0</id>
 			<name></name>
 			<type>30</type>
 			<matcher>

--- a/example/ToolbarAndroidExample-expo-063/android/app/.classpath
+++ b/example/ToolbarAndroidExample-expo-063/android/app/.classpath
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11/"/>
+	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
+	<classpathentry kind="output" path="bin/default"/>
+</classpath>

--- a/example/ToolbarAndroidExample-expo-063/android/app/.project
+++ b/example/ToolbarAndroidExample-expo-063/android/app/.project
@@ -1,10 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>ToolbarAndroidExample</name>
-	<comment>Project android__ created by Buildship.</comment>
+	<name>ToolbarAndroidExampleexpo063-app</name>
+	<comment>Project ToolbarAndroidExampleexpo063-app created by Buildship.</comment>
 	<projects>
 	</projects>
 	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
 		<buildCommand>
 			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
 			<arguments>
@@ -12,11 +17,12 @@
 		</buildCommand>
 	</buildSpec>
 	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
 		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
 	</natures>
 	<filteredResources>
 		<filter>
-			<id>0</id>
+			<id>1630694511165</id>
 			<name></name>
 			<type>30</type>
 			<matcher>

--- a/example/ToolbarAndroidExample-expo-063/android/app/.settings/org.eclipse.buildship.core.prefs
+++ b/example/ToolbarAndroidExample-expo-063/android/app/.settings/org.eclipse.buildship.core.prefs
@@ -1,0 +1,2 @@
+connection.project.dir=..
+eclipse.preferences.version=1

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@react-native-community/toolbar-android",
-  "version": "0.2.0",
+  "version": "0.2.1",
   "description": "A React Native Component for Android Toolbar",
   "keywords": [
     "toolbar-android",

